### PR TITLE
fix: amd64 build and arm build ci

### PR DESCRIPTION
**Pull request by iammadab**
## Issue being fixed or feature implemented

Current CI process fails to build for amd64 and arm

## What was done?

Split up the build process for both architectures
Changed the CC and CXX env variables

## How Has This Been Tested?

Deploy and it built successfully

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

**Review by QuantumExplorer — DISMISSED** [review]

File: .github/workflows/coverage.yml [hunk 1]
```diff
@@ -91,8 +91,19 @@ jobs:
...
       - name: install
+        run: sudo make install
+        if: env.GIT_DIFF != ''
+      - name: build
         run: GOOS=linux GOARCH=${{ matrix.goarch }} make build
-        if: "env.GIT_DIFF != ''"
+        if: matrix.goarch == 'amd64' && env.GIT_DIFF != ''
+      - name: build
+        run: |
+          sudo apt-get install gcc-arm-linux-gnueabihf
```
Comment (shotonoff) [inline .github/workflows/coverage.yml:101]: what do you think groupping deps in one line `sudo apt-get install gcc-arm-linux-gnueabihf g++-8-arm-linux-gnueabihf`?

File: consensus/core_chainlock_test.go [hunk 2]
```diff
@@ -143,6 +143,7 @@ func invalidProposeCoreChainLockFunc(t *testing.T, height int64, round int32, cs
 }
 
 func TestReactorInvalidBlockChainLock(t *testing.T) {
+	// Leads to race, explore
```
Comment (shotonoff) [inline consensus/core_chainlock_test.go:146]: what do you think of to add `TODO` at the beginning?

**Review by shotonoff — APPROVED** [review]
please have a look at the couple of minor suggestions

File: .github/workflows/coverage.yml (hunk 1 above)
Comment (iammadab) [reply .github/workflows/coverage.yml:101]: Cool, would make those changes

**Review by shotonoff — APPROVED** [review]
👍

**Review by shumkov — APPROVED** [review]
LGTM!

**Pull request merged**